`python3Packages.klein` is broken (test failure) due to incompatibility with werkzeug `2.1.2`

`python3Packages.klein` (version `21.8.0`) is broken (test failure) with werkzeug version `2.1.2`, but works with werkzeug `2.1.0`.

#### Affected branches (as of 2022-06-26)
- nixos-22.05
- nixpkgs-unstable

#### Build log
```
python3.9-klein> ===============================================================================
python3.9-klein> [ERROR]
python3.9-klein> Traceback (most recent call last):
python3.9-klein>   File "/nix/store/wn395hf4q94521yvi3v4mqa6lhpkj8ny-python3.9-klein-21.8.0/lib/python3.9/site-packages/klein/test/test_resource.py", line 642, in test_addSlash
python3.9-klein>     request.setHeader.assert_has_calls(
python3.9-klein>   File "/nix/store/7l0dc127v4c2m3yar0bmqy9q6sfmypin-python3-3.9.13/lib/python3.9/unittest/mock.py", line 944, in assert_has_calls
python3.9-klein>     raise AssertionError(
python3.9-klein> builtins.AssertionError: Calls not found.
python3.9-klein> Expected: [call(b'Content-Type', b'text/html; charset=utf-8'),
python3.9-klein>  call(b'Content-Length', b'258'),
python3.9-klein>  call(b'Location', b'http://localhost:8080/foo/')]
python3.9-klein> Actual: [call(b'Content-Type', b'text/html; charset=utf-8'),
python3.9-klein>  call(b'Content-Length', b'239'),
python3.9-klein>  call(b'Location', b'http://localhost:8080/foo/')]
python3.9-klein> klein.test.test_resource.KleinResourceTests.test_addSlash
python3.9-klein> -------------------------------------------------------------------------------
python3.9-klein> Ran 242 tests in 50.854s
python3.9-klein> FAILED (errors=1, successes=241)
```

cc @exarkun (maintainer)

--- a/pkgs/development/python-modules/klein/default.nix
+++ b/pkgs/development/python-modules/klein/default.nix
@@ -1,30 +1,65 @@
-{ stdenv, lib, buildPythonPackage, fetchPypi, python
-, attrs, enum34, hyperlink, incremental, six, twisted, typing, tubes, werkzeug, zope_interface
-, hypothesis, treq
+{ lib
+, stdenv
+, attrs
+, buildPythonPackage
+, fetchFromGitHub
+, hyperlink
+, hypothesis
+, incremental
+, python
+, pythonOlder
+, treq
+, tubes
+, twisted
+, typing-extensions
+, werkzeug
+, zope_interface
 }:
 
 buildPythonPackage rec {
   pname = "klein";
-  version = "21.8.0";
+  version = "unstable-2022-06-26";
+  format = "setuptools";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "1mpydmz90d0n9dwa7mr6pgj5v0kczfs05ykssrasdq368dssw7ch";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "twisted";
+    repo = pname;
+    rev = "d8c2b92a3c77aa64c596696fb6f07172ecf94a74";
+    hash = "sha256-RDZqavkteUbARV78OctZtLIrE4RoYDVAanjwE5i/ZeM=";
   };
 
-  propagatedBuildInputs = [ attrs enum34 hyperlink incremental six twisted typing tubes werkzeug zope_interface ];
+  propagatedBuildInputs = [
+    attrs
+    hyperlink
+    incremental
+    twisted
+    tubes
+    werkzeug
+    zope_interface
+  ] ++ lib.optionals (pythonOlder "3.8") [
+    typing-extensions
+  ];
 
-  checkInputs = [ hypothesis treq ];
+  checkInputs = [
+    hypothesis
+    treq
+  ];
 
   checkPhase = ''
     ${python.interpreter} -m twisted.trial -j $NIX_BUILD_CORES klein
   '';
 
+  pythonImportsCheck = [
+    "klein"
+  ];
+
   meta = with lib; {
-    broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;
     description = "Klein Web Micro-Framework";
-    homepage    = "https://github.com/twisted/klein";
-    license     = licenses.mit;
+    homepage = "https://github.com/twisted/klein";
+    license = licenses.mit;
     maintainers = with maintainers; [ exarkun ];
+    broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;
   };
 }

